### DISCUSSION TITLE
Confusion over process.NextTick in node.

### DISCUSSION THREAD

**Discussion by tmasopust** [libuv/libuv#3986] [Q&A]
The node documentation describes a tick as "Every time the event loop takes a full trip, we call it a tick." https://nodejs.dev/en/learn/understanding-processnexttick/

Is this correct. I've seen presentations from the original developers that imply a tick happens between the loop's stages.

Which one is correct? The only thing I could find on http://docs.libuv.org/en/v1.x/design.html was "The event loop caches the current time at the start of the event loop tick in order to reduce the number of time-related system calls."

**Comment by trevnorris** [comment]
Event loop naming conventions in Node.js are unfortunate and lead to conflicts since they have changed over time.

In the past a **tick** has meant both an event loop **iteration** or **phase** (individual set of operations within a single event loop iteration). The article you reference uses inconsistent definitions. The following makes it sound like a **tick** is an **iteration**:

> Every time the event loop takes a full trip, we call it a tick.

But then it proceeds to describe the execution of `nextTick()` callbacks to be executed at the end of every **phase**.

It's better defined in https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick#processnexttick. If you want an in-depth discussion about how the event loop works, check out https://nodesource.com/blog/event-loop-utilization-nodejs/.
